elastic/beats
# Add support for cloudfoundry logs collection using consumer v1 API

## What does this PR do?

Add support for collection of logs from cloud foundry using consumer v1 API. v1 API uses doppler/trafficcontrol to get events from the firehose, v2 API uses the reverse log proxy (RLP).

v2 API is still supported, and can be selected by adding `version: v2`  to the configuration, v1 is used by default as is in principle more reliable. `version` is ignored by now by Metricbeat, that continues using `v2`. Support for `v1` in Metricbeat will be implemented in a future change.

This change adds some additional dependencies for the doppler consumer (noaa).

## Why is it important?

We already support the v2 API, but we saw that this API is not reliable enough (see #18202).

## Checklist


## Author's Checklist

- [x] Review FIXMEs in code.
- [x] Test with PCF

## How to test this PR locally

- Run filebeat with the cloudfoundry input and check that it collects events. Try these combinations of the `version` option:
  - No `version` option (what would use `v1`)
  - `version: v1`.
  - `version: v2`.
  - For other values of `version` it should fail to start.

## Related issues

- Fixes elastic/beats#18202
- Relates elastic/beats#16520
- Includes integration tests from elastic/beats#19018
